src/canvas-driver.js

```diff
@@ -7,79 +7,114 @@ function renderElement (context, element, parent) {
     parent = {x: 0, y: 0};
   }
 
-  const realX = parent.x + element.x;
-  const realY = parent.y + element.y;
-
+  const origin = {
+    x: element.x ? parent.x + element.x : parent.x,
+    y: element.y ? parent.y + element.y : parent.y
+  };
+  
   if (element.font) {
     context.font = element.font;
   }
 
   if (element.kind === 'rect') {
-    element.draw.forEach(operation => {
-      context.lineWidth = operation.lineWidth || 1;
+    drawRect(context, element, origin);
+  }
+
+  if (element.kind === 'text') {
+    drawText(context, element, origin);
+  }
+
+  if (element.kind === 'line') {
+    drawLine(context, element, origin);
+  }
+
+  element.children && element.children.forEach(child => renderElement(context, child, element))
+}
+
+function drawLine(context, element, origin) {
+  context.lineWidth = element.style.lineWidth || 1;
```

Comment:
If the user calls `line()` without providing a style object, they get this error: `Uncaught TypeError: Cannot read property 'lineWidth' of undefined`.

Could we use `defaults` in the `line` helper, in the same way `text` is done?